cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/FlowUtils.java

```diff
@@ -68,8 +69,13 @@ public static long generateConsumerGroupId(Program program, String flowletId) {
    * Generates a queue consumer groupId for the given flowlet in the given program id.
    */
   public static long generateConsumerGroupId(Id.Program program, String flowletId) {
+    // Use 'developer' in place of a program's namespace for programs in the 'default' namespace
+    // to support backwards compatibility for queues and streams.
+    String namespace = program.getNamespaceId();
+    String backwardsCompatibleNamespace =
+      Constants.DEFAULT_NAMESPACE.equals(namespace) ? Constants.DEVELOPER_ACCOUNT : namespace;
     return Hashing.md5().newHasher()
-                  .putString(program.getNamespaceId())
+                  .putString(backwardsCompatibleNamespace)
```

Comment:
namespace is not required in the groupId hash, since the queue/stream tables are created in separate namespaces. (The table name includes the namespace).

The hash only includes namespace ('developer' for default namespace) for backwards compatibility).